<launch>
  <node name="RosAria" pkg="rosaria" type="RosAria" />
  <node name="urg_node" pkg="urg_node" type="urg_node" >
    <param name="base_frame" value="scan" />
  </node>
  <node name="pioneer_tf" pkg="javis_pkg" type="javis_pkg_broadcastor" />
  <node name="gmapping" pkg="gmapping" type="slam_gmapping" >
    <param name="scan_frame" value="/scan"/>
    <param name="odom_frame" value="/odom"/>
    <param name="base_frame" value="/base_link"/>
    <param name="map_frame" value="/map"/>
    <param name="map_update_interval" value="6.0"/>
    <param name="maxUrange" value="6.0"/>
    <param name="sigma" value="0.05"/>
    <param name="kernelSize" value="1"/>
    <param name="lstep" value="0.05"/>
    <param name="astep" value="0.05"/>
    <param name="iterations" value="5"/>
    <param name="lsigma" value="0.075"/>
    <param name="ogain" value="3.0"/>
    <param name="lskip" value="0"/>
    <param name="srr" value="0.01"/>
    <param name="srt" value="0.02"/>
    <param name="str" value="0.01"/>
    <param name="stt" value="0.02"/>
    <param name="linearUpdate" value="0.25"/>
    <param name="angularUpdate" value="0.25"/>
    <param name="temporalUpdate" value="-1.0"/>
    <param name="resampleThreshold" value="0.5"/>
    <param name="particles" value="300"/>
    <param name="xmin" value="-10.0"/>
    <param name="ymin" value="-10.0"/>
    <param name="xmax" value="10.0"/>
    <param name="ymax" value="10.0"/>
    <param name="delta" value="0.05"/>
    <param name="llsamplerange" value="0.01"/>
    <param name="llsamplestep" value="0.01"/>
    <param name="lasamplerange" value="0.005"/>
    <param name="lasamplestep" value="0.005"/>
  </node>
  <node name="teleop_keyboard" pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" output="screen" >
    <param name="_linear" value="0.1" type="double"/>
    <param name="_angular" value="0.5" type="double"/>
    <remap from="/cmd_vel" to="RosAria/cmd_vel"/>
  </node>
  <!--
  <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
    <param name="odom" value="/RosAria/pose"/>
    <param name="output_frame" value="/odom_frame"/>
    <param name="freq" value="10.0"/>
    <param name="sensor_timeout" value="10.0"/>
    <param name="odom_used" value="true"/>
    <param name="imu_used" value="false"/>
    <param name="vo_used" value="false"/>
    <param name="debug" value="false"/>
    <param name="self_diagnose" value="true"/>
  </node>
  -->
  <node name="rviz" pkg="rviz" type="rviz"/>
</launch>


